### PR TITLE
Huge pages value is not reflecting in 'nr_hugepages'

### PR DESCRIPTION
**Pull request by nasastry**
Huge pages value is not reflecting in
'/proc/sys/vm/nr_hugepages'

1. While reading the value from nr_hugepages file, new line need
   to be handled
2. File open and close need to be handled properly.

This patch addresses the above issues.

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

File: virttest/test_setup.py [hunk 1]
```diff
@@ -487,18 +487,20 @@ def set_hugepages(self):
         error_context.context(
             "setting hugepages limit to %s" % self.target_hugepages)
         hugepage_cfg = open(self.kernel_hp_file, "r+")
-        hp = hugepage_cfg.readline()
+        hp = hugepage_cfg.readline().strip()
+        hugepage_cfg.close()
         while int(hp) < self.target_hugepages:
             loop_hp = hp
+            hugepage_cfg = open(self.kernel_hp_file, "r+")
```
Comment (balamuruhans) [inline virttest/test_setup.py]: it is better to use try and except with IOError and raise exceptions.TestError()

File: virttest/test_setup.py [hunk 2]
```diff
@@ -487,18 +487,20 @@ def set_hugepages(self):
         error_context.context(
             "setting hugepages limit to %s" % self.target_hugepages)
         hugepage_cfg = open(self.kernel_hp_file, "r+")
```
Comment (balamuruhans) [inline virttest/test_setup.py]: same here

File: virttest/test_setup.py [hunk 3]
```diff
@@ -486,19 +486,28 @@ def set_hugepages(self):
...
+            try:
+                hugepage_cfg = open(self.kernel_hp_file, "r+")
+                hugepage_cfg.write(str(self.target_hugepages))
+                hugepage_cfg.flush()
+                hugepage_cfg.seek(0)
+                hp = int(hugepage_cfg.readline().strip())
+                hugepage_cfg.close()
+            except IOError:
+                msg = "Can't read/write from kernel hugepage file"
+                raise exceptions.TestSetupFail(msg)
```
Comment (PandaWei) [inline virttest/test_setup.py]: Do we need open/close the nr_hugepages twice?

Comment (nasastry) [inline virttest/test_setup.py]: I have tried with out closing, file is not reflecting the value user try to put in.

**Comment by nasastry** [comment]
@will-Do Can you please help me in merging this patch. Thanks in advance.

**Comment by xutian** [comment]
Looks OK to me. Merged, thanks @nasastry

**Pull request merged**